src/xbot/common/math/PID.java

```diff
@@ -23,6 +28,37 @@ public void reset()
     {
         m_prevError = 0;
         m_totalError = 0;
+        errorIsSmall = false;
+        derivativeIsSmall = false;
+    }
+    
+    /**
+     * Set how close the error can be before it is considered
+     * "on-target."
+     *            
+     * This is in the same units as your current and goal values.
+     */
+    public void setErrorTolerance(double errorTolerance) {
+        setTolerances(errorTolerance, this.derivativeTolerance);
```

Comment:
This should just set the property manually. You aren't gaining anything by calling the other method, and this adds an extra layer of indirection for a reader. Additionally, if we later wanted to add logic to `setTolerances` that reset things when the value was changed, this would make that logic needlessly complicated.